server/tests/classes.test.ts

```diff
@@ -0,0 +1,518 @@
+
+            expect(response.status).toBe(401)
+        })
+
+        it('Should return unauthorized error', async () => {
+            findClassByIdMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+            const response = await request(app)
+                .get(ClassEndpoint)
+                .set('Authorization', `Bearer ${UNAUTHORIZED_JWT2}`)
+
+            expect(response.status).toBe(401)
+        })
+
+        it('Should return not found when class is not found', async () => {
+            findClassByIdMock.mockImplementation(async () =>
+                Promise.resolve()
+            )
+            const response = await request(app)
+                .get(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+
+            expect(response.status).toBe(404)
+        })
+
+        it('Should return internal server error', async () => {
+            findClassByIdMock.mockImplementation(async () =>
+                Promise.reject('Internal server error')
+            )
+            const response = await request(app)
+                .get(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+
+            expect(response.status).toBe(500)
+        })
+    })
+
+    describe('POST /classes/course/:courseId', () => {
+        let createClassMock: jest.SpyInstance
+        beforeAll(() => {
+            createClassMock = jest.spyOn(Class.prototype, 'save')
+            jest.mock('multer', () => mockMulter)
+        })
+
+        it('Should return OK when class is created', async () => {
+            createClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+            const response = await request(app)
+                .post(CourseClassesEndpoint)   
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1')
+                .field('order', 1)
+                .field('description', 'Descripción 1')
+                //Generar archivo de video con mock
+                .attach('file', Buffer.from(''), {
+                    contentType: 'video/mp4',
+                    filename: 'mockedFile1.mp4',
+                  });
+            expect(response.status).toBe(201)
+            expect(response.body.message).toBe('Class created successfully')
+        })
+
+        it('Invalid data', async () => {
+            createClassMock.mockImplementation(async () => {
+                throw new Error('Invalid data')
+            })
+
+            const response = await request(app)
+                .post(CourseClassesEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1')
+                .field('order', 1)
+                .field('description', 'Descripción 1')
+                .attach(
+                    'file',
+                    // TODO: Change to real video file
```

Comment:
Delete this comment please. I understand that 'TODO' has already been done, although it is still better to put an mp4 file in the buffer, for code coherence